src/rfbzero/experiment.py

```diff
@@ -31,46 +31,72 @@ class CyclingProtocolResults:
     def __init__(self, duration: float, time_increment: float, charge_first: bool = True) -> None:
         self.duration = duration
         self.time_increment = time_increment
-        self.size = int(duration / time_increment)
         self.charge_first = charge_first
         self.compute_soc = True
 
-        self.step = 0
-        self.step_time = [0.0] * self.size
-        self.step_is_charge = [False] * self.size
-
-        self.current = [0.0] * self.size
-        self.cell_v = [0.0] * self.size
-        self.ocv = [0.0] * self.size
-
-        self.c_ox_cls = [0.0] * self.size
-        self.c_red_cls = [0.0] * self.size
-        self.c_ox_ncls = [0.0] * self.size
-        self.c_red_ncls = [0.0] * self.size
-        self.delta_ox = [0.0] * self.size
-        self.delta_red = [0.0] * self.size
-        self.soc_cls = [0.0] * self.size
-        self.soc_ncls = [0.0] * self.size
-
-        self.act = [0.0] * self.size
-        self.mt = [0.0] * self.size
-        self.loss = [0.0] * self.size
-
-        # Total number of cycles is unknown at start, thus sizes are undetermined
-        self.half_cycles = 0
+        #: The maximum number of time steps that could be performed by the simulation.
```

Comment:
this and the line below it essentially say the same thing. 
`max_steps` is what the user wants, `step` is what the user gets